dali/base/dadfs.cpp

```diff
@@ -4941,27 +4949,39 @@ protected: friend class CDistributedFilePart;
         double fileAgeDays = difftime(time(nullptr), dt.getSimple())/(24*60*60);
         double sizeGB = getDiskSize(true, false) / ((double)1024 * 1024 * 1024);
         const IPropertyTree *attrs = root->queryPropTree("Attr");
+        bool doLegacyAccessCostCalc = false;
         __int64 numDiskWrites = 0, numDiskReads = 0;
         if (attrs)
         {
-            numDiskWrites = attrs->getPropInt64("@numDiskWrites");
-            numDiskReads = attrs->getPropInt64("@numDiskReads");
+            if (hasReadWriteCostFields(attrs))
+            {
+                // Newer files have readCost and writeCost attributes
+                accessCost = cost_type2money(attrs->getPropInt64(getDFUQResultFieldName(DFUQRFreadCost)) + attrs->getPropInt64(getDFUQResultFieldName(DFUQRFwriteCost)));
+            }
+            else
+            {
+                // (Costs needs to be calculated from numDiskReads and numDiskWrites for legacy files)
+                numDiskWrites = attrs->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskWrites));
+                numDiskReads = attrs->getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads));
+                doLegacyAccessCostCalc = true;
```

Comment:
Possibly only for files, not indexes if the index stats are currently wildly wrong.